infermedica_api/connectors/common.py

```diff
@@ -40,12 +40,9 @@ class SearchConceptType(Enum):
     RISK_FACTOR = "risk_factor"
     LAB_TEST = "lab_test"
 
-    @staticmethod
-    def has_value(val: Union["SearchConceptType", str]) -> bool:
-        try:
-            return val in SearchConceptType
-        except TypeError:
-            return val in (item.value for item in SearchConceptType)
+    @classmethod
+    def has_value(cls, value: Union["SearchConceptType", str]) -> bool:
+        return value in (val.value for val in cls.__members__.values())
```

Comment:
Unfortunately new code does not handle all required cases. The function is suppose to verify if a string value or a enum value is a part of this enum (see allow value types for `value`).
```
>>> class SearchConceptType(Enum):
...     """Enum to hold search filter constants."""
...     SYMPTOM = "symptom"
...     RISK_FACTOR = "risk_factor"
...     LAB_TEST = "lab_test"
...     
...     @classmethod
...     def has_value(cls, value: Union["SearchConceptType", str]) -> bool:
...         return value in (val.value for val in cls.__members__.values())
... 
>>> 
>>> SearchConceptType.has_value("s")
False
>>> SearchConceptType.has_value("symptom")
True
>>> SearchConceptType.has_value(SearchConceptType.SYMPTOM)
False  <-- Should be True
```

That is why we previously had to cases one in `try` and different in `except`.

Reply:
imho there should be `SearchConceptType.SYMPTOM.value` but nevertheless I'll change it and make it bulletproof...

Reply:
Done
<img width="769" alt="Screenshot 2021-03-26 at 16 23 22" src="https://user-images.githubusercontent.com/6198355/112654351-96e3bc80-8e4f-11eb-8c33-6ae067725eb4.png">

Reply:
Looks better now thanks, maybe instead of almost repeated code we can do:
```
if isinstance(value, SearchConceptType):
    return val in SearchConceptType
return val in (item.value for item in SearchConceptType)
```
So we will just replace `try/catch` with `if isinstance(value, SearchConceptType):`. Notice that we are also not accessing a private method while verifying if string is present.

Reply:
@akszydelko done
<img width="646" alt="Screenshot 2021-03-29 at 10 13 25" src="https://user-images.githubusercontent.com/6198355/112807613-63d04180-9078-11eb-9c12-d391ef80125e.png">